components/Credit/Address.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { ChangeEvent, useCallback, useEffect, useState } from 'react';
+import Link from 'next/link';
+import { type Address, isAddress, getAddress } from 'viem';
+import { useTranslation } from 'react-i18next';
+import ContentPasteIcon from '@mui/icons-material/ContentPaste';
+import {
+  Avatar,
+  Box,
+  Button,
+  IconButton,
+  OutlinedInput,
+  Step,
+  StepContent,
+  StepLabel,
+  Stepper,
+  Typography,
+} from '@mui/material';
+import { App } from '.';
+
+type Props = {
+  value?: Address;
+  app: App;
+  onNextStep: () => void;
+  onChange: (a?: Address) => void;
+};
+
+const Address = ({ value, app, onNextStep, onChange }: Props) => {
+  const [hasOpenedApp, setHasOpenedApp] = useState(false);
+  const [address, setAddress] = useState<string | undefined>();
+  const { t } = useTranslation();
+  useEffect(() => setAddress(value), [value]);
```

Comment:
no, because `value` changes, useState init only happens once. I know the state of the address in the component looks confusing but basically the 'real' saved address is saved one level above and the `address` in the component only handles the text input.